Tests/CartonCommandTests/InitCommandTests.swift

```diff
@@ -0,0 +1,116 @@
+// Copyright 2020 Carton contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Created by Cavelle Benjamin on Dec/20/20.
+//
+
+@testable import CartonCLI
+import TSCBasic
+import XCTest
+
+extension InitCommandTests: Testable {}
+
+final class InitCommandTests: XCTestCase {
+  func testDefaultArgumentParsing() throws {
+    // given
+    let arguments: [String] = []
+
+    // when
+
+    AssertParse(Dev.self, arguments) { command in
+      // then
+      XCTAssertNotNil(command)
+    }
+  }
+
+  func testHelpString() throws {
+    // given
+    let expectation =
+      """
+      OVERVIEW: Create a Swift package for a new SwiftWasm project.
+
+      USAGE: carton init [--template <template>] [--name <name>] <subcommand>
+
+      OPTIONS:
+        --template <template>   The template to base the project on.
+        --name <name>           The name of the project
+        --version               Show the version.
+        -h, --help              Show help information.
+
+      SUBCOMMANDS:
+        list-templates          List the available templates
+
+        See 'carton help init <subcommand>' for detailed help.
+      """
+    // when
+    // then
+
+    AssertExecuteCommand(command: "carton init -h", expected: expectation)
+  }
+
+  func testWithNoArguments() throws {
```

Comment:
The main responsibility of `carton init` command is generating a package, so I think we should ensure that there are expected files in the package directory instead of just checking stdout.

Reply:
That is fair and a fairly straight forward change.